### PR TITLE
userguide: add section about exception policy - v3

### PR DESCRIPTION
**Pull request by jufajardini**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5475
https://redmine.openinfosecfoundation.org/issues/5515

Describe changes:
- add a configuration section describing the Exception Policies
- minor typo fixes and rewordings
- incorporate what was in https://github.com/OISF/suricata/pull/7873

Updates from last PR (https://github.com/OISF/suricata/pull/7886 ):
- add reference to the suricata.yaml documentation
- some rewordings

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW1_stats_chk | | |
| .tcp.rst | 126232 | 102722 | 81.38% |
| | SURI_TLPR1_stats_chk | | |
| .flow.memuse | 562615488 | 607323328 | 107.95% |
| .app_layer.error.http.parser | 1548 | 1103 | 71.25% | 

Pipeline 9371

File: doc/userguide/configuration/exception-policies.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
+.. _exception policies:
+
+Exception Policies
+==================
+
+Suricata has a set of configuration variables to indicate what should the engine
+do when certain exception conditions, such as hitting a memcap, are reached.
+
+They are called Exception Policies and are configurable via suricata.yaml. If
+enabled, the engine will call them when it  reaches exception states.
```
Comment (jlucovsky) [inline doc/userguide/configuration/exception-policies.rst:10]: nit: s/it. reaches/it reaches/ (one space)

File: doc/userguide/configuration/exception-policies.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+For developers or for researching purposes, there are also simulation options
+exposed in debug mode and passed via command-line. These exist to force or
+simulate failures or errors and understand Suri behavior under such conditions.
```
Comment (jlucovsky) [inline doc/userguide/configuration/exception-policies.rst:14]: s/Suri/Suricata/

**Comment by jufajardini** [comment]
V4 incoming. :D

File: doc/userguide/configuration/exception-policies.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+documentation).
+
+The possible values for the exception policies, and the resulting behaviors,
+are:
+
+- ``drop-flow``: disable inspection for the whole flow (packets, payload,
+  application layer protocol), drop the packet and all future packets in the
+  flow.
+- ``drop-packet``: drop the packet.
+- ``reject``: same as ``drop-flow``, but reject the first packet as well.
```
[the remark this answers is not in the document]
Comment (jufajardini) [reply doc/userguide/configuration/exception-policies.rst:63]: indeed '-'

**Pull request closed without merging**